language-plutus-core/src/Language/PlutusCore/Rename.hs

```diff
@@ -11,26 +11,27 @@ import           Language.PlutusCore.Name
 import           Language.PlutusCore.Quote
 import           Language.PlutusCore.Rename.Internal
 import           Language.PlutusCore.Type
+import           PlutusPrelude
 
 -- | The class of things that can be renamed.
 -- I.e. things that are capable of satisfying the global uniqueness condition.
 class Rename a where
     -- | Rename 'Unique's so that they're globally unique.
     -- In case there are any free variables, they must be left untouched.
     -- Must always assign new names to bound variables,
-    -- so that @rename@ can be used for alpha renaming as well.
+    -- so that @rename@ can be used for alpha-renaming as well.
     rename :: MonadQuote m => a -> m a
 
 instance HasUnique (tyname ann) TypeUnique => Rename (Type tyname ann) where
-    rename = runDirectRenameM . renameTypeM
+    rename = through markNonFreshType >=> runDirectRenameM . renameTypeM
```

Comment:
:+1:

You can delete the one use of these functions directly, which is in `test/Check/Spec.hs` since it uses `rename` so is now correct anyway.

Reply:
Done.